### PR TITLE
making histolab pip-installable

### PR DESCRIPTION
**Pull request by ernestoarbitrio**
- setup.py file and all related stuff
- requirements refactoring

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    "scipy==1.4.1",
+]
+
+test_requires = ["pytest", "coverage", "pytest-cov", "coveralls", "pytest-mpl"]
+
+setuptools.setup(
+    name="histolab",
+    version=version,
+    maintainer="Histolab Developers",
+    mantainer_email="ernesto.arbitrio@gmail.com",
```
Comment (alessiamarcolini) [inline setup.py:43]: typo:
```suggestion
    maintainer_email="ernesto.arbitrio@gmail.com",
```

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+coverage:
+	pytest --cov-report term-missing --cov=src --cov=tests
+
+dist:
+	rm -rf dist/
+	$(SETUP) sdist bdist_wheel
+
+docs:
+	$(MAKE) -C docs html
```
Comment (alessiamarcolini) [inline Makefile:34]: Maybe I am missing something, but where is Sphinx configured? I mean, is this enough?

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+opendocs:
+	open docs/build/html/index.html
+
+unit-coverage:
+	pytest --cov-report term-missing --cov=src tests/unit
+
+upload:
+	rm -rf dist/
+	$(SETUP) sdist bdist_wheel
+	twine upload dist/*
```
Comment (alessiamarcolini) [inline Makefile:45]: I don't see twine in any requirements file

File: Makefile (hunk 2 above)
Comment (ernestoarbitrio) [reply Makefile:34]: it's for future, so i have in mind that we wanna have sphinx and when we wanna release a new doc version we already have the make command to do that

File: setup.py (hunk 1 above)
Comment (ernestoarbitrio) [reply setup.py:43]: yes it's a typo and BTW i don't wanna use my email, do we wanna create a project discussion list ? eg. histolab-discussions@python.org ? XD

Comment (alessiamarcolini) [reply setup.py:43]: Oh man, you're thinking so big :D 

I think this is a bit overkill for the moment... @nicolebussola what do you think?

File: Makefile (hunk 2 above)
Comment (alessiamarcolini) [reply Makefile:34]: Yes we want to have sphinx for sure! But having the command here suggests that NOW it's possible to build docs. Maybe we can comment `$(MAKE) -C docs html` and have something like `@echo "Feature not implemented yet"`

File: setup.py (hunk 1 above)
Comment (ernestoarbitrio) [reply setup.py:43]: as per discussion, but let leave my email for the moment

**Comment by alessiamarcolini** [comment]
LGTM!

**Pull request merged**